numpy/core/src/common/simd/avx2/arithmetic.h

```diff
@@ -127,6 +135,19 @@ NPY_FINLINE npy_uint32 npyv_sum_u32(__m256i a)
     return _mm_cvtsi128_si32(s1);
 }
 
+NPY_FINLINE npy_uint32 npyv_sum_u16(__m256i a)
+{
+    npyv_u32x2 res = npyv_expand_u32_u16(a);
+    return (unsigned)npyv_sum_u32(_mm256_add_epi32(res.val[0], res.val[1]));
+}
```

Comment:
```suggestion
NPY_FINLINE npy_uint32 npyv_sumup_u16(npyv_u16 a)
{
    const npyv_u16 even_mask = _mm256_set1_epi32(0x0000FFFF);
    __m256i even  = _mm256_and_si256(a, even_mask);
    __m256i odd   = _mm256_srli_epi32(a, 16);
    __m256i eight = _mm256_add_epi32(even, odd);
    return npyv_sum_u32(eight);
}
```
expansion even and odd elements by shift and bitwise-and should perform faster than halves expansion (`_mm256_cvtepu16_epi32`)